src/lib/fetch.js

```diff
@@ -16,20 +16,26 @@ import constants from "../constants";
  */
 export const getUserData = async (uid = "", includePrograms = false) => {
   const getUserDataEndpoint = (uid = "", includePrograms = false) =>
-    `${constants.SERVER_URL}/getUserData/${uid}${includePrograms ? "?programs=true" : ""}`;
+    `${constants.SERVER_URL}/user/get?uid=${uid}${includePrograms ? "&programs=true" : ""}`;
 
   const options = {
     method: "get",
     mode: "cors", // no-cors, cors, *same-origin
   };
 
   try {
-    let result = await fetch(getUserDataEndpoint(uid, includePrograms), options);
-    let { ok, data, error } = await result.json();
-
+    const result = await fetch(getUserDataEndpoint(uid, includePrograms), options);
+    let ok = await result.ok;
+    if (!ok) {
```

Comment:
Is it possible to check for the "user does not exist" error specifically before trying to create a new user?

Reply:
Yeah we can check it via the response status (404 => user doesn't exist). Nice idea.